src/main/java/com/metamx/metrics/KeyedDiff.java

```diff
@@ -41,7 +41,7 @@
 
   public static Map<String, Long> subtract(Map<String, Long> xs, Map<String, Long> ys)
   {
-    assert xs.keySet() == ys.keySet();
+    assert xs.keySet().equals(ys.keySet());
```

Comment:
@logarithm Moreover this is in an `assert` statement so it is not executed in production at all.